### PR TITLE
[Python API] Added exception handling for 'None' model argument in save_model function

### LINKED ISSUE #21086: [Good First Issue]: `openvino.save_model(None, ...)` crashes instead of raising an exception
### Context

We strive to have a clear, user friendly API in OpenVINO. An issue has been found where when giving `None` as an argument to `openvino.save_model()` function, the result is a segmentation fault, instead of an exception.

Reproducer:
```bash
>>> import openvino as ov
>>> ov.save_model(None, 'model.xml')
Segmentation fault (core dumped)
``` 

The binding can be found [here](https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/src/pyopenvino/pyopenvino.cpp#L163-L193).

### What needs to be done?

`save_model` function binding needs to be modified in a way which will check the type of `model` argument. If it's None, a proper Python exception should be raised. 

Tip:
Please use pybind functionalities for [types](https://pybind11.readthedocs.io/en/stable/advanced/pycpp/object.html) and [errors](https://pybind11.readthedocs.io/en/stable/advanced/exceptions.html)for implementation, for example:
- `py::None`
- `py::attribute_error`

### Example Pull Requests

--

### Resources

- [What is OpenVINO?](https://github.com/openvinotoolkit/openvino#what-is-openvino-toolkit)
- [Contribution guide](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [Blog post on contributing to OpenVINO](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [User documentation](https://docs.openvino.ai/)
- [Python bindings docs](https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/docs/code_examples.md)

### Contact points

@openvinotoolkit/openvino-ie-python-api-maintainers 

### Ticket

125053

### PR DESCRIPTION
**Pull request by sydarb**
### Details:
 - Modified the `openvino.save_model()` function binding to perform a type check on the 'model' argument. If the argument is `None`, an `AttributeError` exception is now raised instead of segmentation fault.

Screenshot 2023-12-13 at 2 15 58 AM

### Tickets:
 - Closes #21086

### CODE REVIEW

**Comment by akuporos** [comment]
Hello @sydarb,
Thank you for your contribution!
Could you please also add a test to [this file](https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/tests/test_runtime/test_model.py#L222)?
[Here ](https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/docs/test_examples.md#writing-openvino-python-api-tests) you may find how to test and add tests.

Thank you,
Anastasia

File: src/bindings/python/src/pyopenvino/pyopenvino.cpp [hunk 1]
```diff
@@ -165,6 +165,9 @@ PYBIND11_MODULE(_pyopenvino, m) {
         [](std::shared_ptr<ov::Model>& model,
            const py::object& xml_path,
            bool compress_to_fp16) {
+            if (model == nullptr) {
+                throw py::attribute_error("'model' argument is required and cannot be None.");
+            }
```
Comment (akuporos) [inline src/bindings/python/src/pyopenvino/pyopenvino.cpp:170]:
```suggestion
           if (model == nullptr) {
               throw py::attribute_error("'model' argument is required and cannot be None.");
           }
```

File: src/bindings/python/tests/test_runtime/test_model.py [hunk 2]
```diff
@@ -566,3 +567,8 @@ def test_model_add_remove_variable():
     assert variable_by_id.info.variable_id == "var_id_667"
     model.remove_variable(variable_1)
     assert len(model.get_variables()) == 1
+
+def test_save_model_with_none():
```
Comment (akuporos) [inline src/bindings/python/tests/test_runtime/test_model.py:571]:
```suggestion


def test_save_model_with_none():
```

Comment (akuporos) [reply src/bindings/python/tests/test_runtime/test_model.py:571]: to fix codestyle :)

**Comment by p-wysocki** [comment]
build_jenkins

**Review by akuporos — APPROVED** [review]
LGTM

File: src/bindings/python/tests/test_runtime/test_model.py [hunk 3]
```diff
@@ -566,3 +567,9 @@ def test_model_add_remove_variable():
     assert variable_by_id.info.variable_id == "var_id_667"
     model.remove_variable(variable_1)
     assert len(model.get_variables()) == 1
+
+
+def test_save_model_with_none():
+    with pytest.raises(AttributeError) as e:
+        save_model(model=None, output_model='model.xml')
```
Comment (p-wysocki) [inline src/bindings/python/tests/test_runtime/test_model.py:574]:
```suggestion
        save_model(model=None, output_model="model.xml")
```
Linter fix

**Comment by p-wysocki** [comment]
build_jenkins

**Comment by p-wysocki** [comment]
I enabled auto merge. Thank you very much for your contribution @sydarb!

**Pull request merged**